packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -721,16 +721,16 @@ class _CupertinoEdgeShadowDecoration extends Decoration {
 
   @override
   _CupertinoEdgeShadowDecoration lerpFrom(Decoration a, double t) {
-    if (a is! _CupertinoEdgeShadowDecoration)
-      return _CupertinoEdgeShadowDecoration.lerp(null, this, t);
-    return _CupertinoEdgeShadowDecoration.lerp(a, this, t);
+    if (a is _CupertinoEdgeShadowDecoration)
+      return _CupertinoEdgeShadowDecoration.lerp(a, this, t);
+    return _CupertinoEdgeShadowDecoration.lerp(null, this, t);
```

Comment:
Do these have to be changed?  If not, should we move this to a separate patch?

Reply:
By changing the order `a` can be promoted to `_CupertinoEdgeShadowDecoration` and avoid an explicit cast if the order remain the same.

Note that with changes regarding type promotion in NNBD the previous code would have working without changes.